src/sample/models/Bot.java

```diff
@@ -31,11 +31,101 @@ public void Move(MoveEnum direction)
     public void MoveRandomly()
     {
         // Determine which moves are possible, then add them to the list
+        getPreferredMoves();
         getPossibleMoves();
         Random randomNumberGenerator = new Random(System.currentTimeMillis());
-        int randomMovementNumber = randomNumberGenerator.nextInt(possibleMoves.size());
-        MoveEnum chosenMove = possibleMoves.get(randomMovementNumber);
-        moveTheBot(chosenMove);
+        if (preferredMoves.size() != 0)
+        {
+            int randomMovementNumber = randomNumberGenerator.nextInt(preferredMoves.size());
+            MoveEnum chosenMove = preferredMoves.get(randomMovementNumber);
+            if (possibleMoves.contains(chosenMove)) {
+                moveTheBot(chosenMove);
+            }
+            preferredMoves.clear();
+        }
+        else
+            {
+                int randomMovementNumber = randomNumberGenerator.nextInt(possibleMoves.size());
+                MoveEnum chosenMove = possibleMoves.get(randomMovementNumber);
+                moveTheBot(chosenMove);
+            }
+    }
+
+    private void getPreferredMoves()
+    {
+        if(this.y != 0)
+        {
+            if (movesMade.size() == 0)
+            {
+                preferredMoves.add(MoveEnum.Up);
+            }
+            for (String move : movesMade)
```

Comment:
Not a big fan of this, I think we need to rethink this.  If down the road we change how moves are stored, we have to completely rework this

Reply:
I can certainly try to simplify this in implementation but it would be local method refactor and I am not sure if you are suggesting or referring to a new class or something different
Thoughts?

Reply:
I was thinking a Hashmap<Integer, Integer> but then I was struggling how to pull the coordinates
I will try to simplify this down and also extract it out to a function